00_Base/src/services/commands.service.ts

```diff
@@ -49,22 +59,82 @@ export class CommandsService {
     }
   }
 
-  private handleCancelReservation(
-    _cancelReservation: CancelReservation,
-  ): OcpiResponse<CommandResponse> {
-    return ResponseGenerator.buildGenericClientErrorResponse({
-      result: CommandResponseType.NOT_SUPPORTED,
-      timeout: this.TIMEOUT,
-    });
+  private async handleCancelReservation(
+    cancelReservation: CancelReservation,
+    fromCountryCode: string,
+    fromPartyId: string,
+  ): Promise<OcpiResponse<CommandResponse>> {
+    try {
+      await this.commandExecutor.executeCancelReservation(
+        cancelReservation,
+        fromCountryCode,
+        fromPartyId,
+      );
+      return ResponseGenerator.buildGenericSuccessResponse({
+        result: CommandResponseType.ACCEPTED,
+        timeout: this.TIMEOUT,
+      });
+    } catch (e) {
+      if (e instanceof NotFoundError) {
+        return ResponseGenerator.buildGenericClientErrorResponse(
+          {
+            result: CommandResponseType.REJECTED,
+            timeout: this.TIMEOUT,
+          },
+          e.message,
+          e as NotFoundError,
+        );
+      } else {
+        console.error(e);
+        return ResponseGenerator.buildGenericServerErrorResponse(
+          {
+            result: CommandResponseType.REJECTED,
+            timeout: this.TIMEOUT,
+          },
+          undefined,
+          e as Error,
+        );
+      }
+    }
   }
 
-  private handleReserveNow(
-    _reserveNow: ReserveNow,
-  ): OcpiResponse<CommandResponse> {
-    return ResponseGenerator.buildGenericClientErrorResponse({
-      result: CommandResponseType.NOT_SUPPORTED,
-      timeout: this.TIMEOUT,
-    });
+  private async handleReserveNow(
+    reserveNow: ReserveNow,
+    fromCountryCode: string,
+    fromPartyId: string,
+  ): Promise<OcpiResponse<CommandResponse>> {
+    try {
+      await this.commandExecutor.executeReserveNow(
+        reserveNow,
+        fromCountryCode,
+        fromPartyId,
+      );
+      return ResponseGenerator.buildGenericSuccessResponse({
+        result: CommandResponseType.ACCEPTED,
+        timeout: this.TIMEOUT,
+      });
+    } catch (e) {
+      if (e instanceof NotFoundError || e instanceof BadRequestError) {
```

Comment:
just curious, is there a reason these two errors are combined together to return a not found error? is there just no handling of a `BadRequestError` currently 🤔

Reply:
Good catch! I corrected the code.